src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php

```diff
@@ -81,6 +81,10 @@ public function onKernelRequest(GetResponseEvent $event)
         if (!$event->isMasterRequest() || '/' !== $request->getPathInfo()) {
             return;
         }
+        // Ignore requests from referrers with the same HTTP host
```

Comment:
@voronkovich let's describe it a bit. What about:

> Ignore requests from referrers with the same HTTP host in order to prevent changing language for users who already selected it in this application.

How does it sound for you?